src/main/java/net/openhft/chronicle/bytes/MappedBytesStore.java

```diff
@@ -479,19 +486,26 @@ public void syncMode(SyncMode syncMode) {
      * @param position to sync with the syncMode()
      */
     public void syncUpTo(long position) {
-        if (syncMode == SyncMode.NONE || address == 0 || refCount() <= 0)
+        syncUpTo(position, this.syncMode);
+    }
+
+    /**
+     * Synchronise from the last complete page up to this position.
+     *
+     * @param position to sync with the syncMode()
+     * @param syncMode to use
+     */
+    public void syncUpTo(long position, SyncMode syncMode) {
+        if (syncMode == SyncMode.NONE || address == 0 || refCount() <= 0 || !OS.isLinux())
             return;
-        long length = position - start;
-        if (length <= syncLength)
+        long positionFromStart = Math.min(safeLimit, position) - start;
```

Comment:
Hi @peter-lawrey if you have time it would be good to get a new test around this (unless its covered already)

Reply:
Good catch, more tests needed.